standard/structs.md

```diff
@@ -60,6 +62,29 @@ A readonly struct has the following constraints:
 
 When an instance of a readonly struct is passed to a method, its `this` is treated like an `in` argument/parameter, which disallows write access to any instance fields (except by constructors).
 
+### §ref-modifier-new-clause Ref modifier
+
+The `ref` modifier indicates that the *struct_declaration* declares a type whose instances are allocated on the execution stack. These types are are called ***ref struct*** types. The `ref` modifier declares that instances may contain ref-like fields, and may not be copied out of its safe-context (§safe-context-rules). The rules for determining the safe context of a ref struct are described in §safe-context-rules.
+
+It is a compile-time error if a ref struct type is used in any of the following contexts:
+
+- As the element type of an array.
+- As the declared type of a field of a class or a non-ref struct.
+- Being boxed to `System.ValueType` or `System.Object`:
+- As a type argument.
+- An async method.
+- An iterator.
+
+> *Note*: A `ref struct` shall not declare `async` instance methods nor use a `yield return` or `yield break` statement within an instance method, because the implicit `this` parameter cannot be used in those contexts. *end note*
+
+- There is no conversion from a `ref struct` type to the type `object` or the type `System.ValueType`.
```

Comment:
This list doesn't have any introduction, which looks a bit odd. I think we could probably either make it "not a list", or add an introductory sentence: "The following additional constraints exist for `ref struct` types:"

Reply:
It's all one list, with the note splitting it. I moved the note below the entire list.